src/bitops.c

```diff
@@ -491,12 +491,10 @@ int checkSignedBitfieldOverflow(int64_t value, int64_t incr, uint64_t bits, int
     int64_t max = (bits == 64) ? INT64_MAX : (((int64_t)1 << (bits - 1)) - 1);
     int64_t min = (-max) - 1;
 
-    /* Note that maxincr and minincr could overflow, but we use the values
-     * only after checking 'value' range, so when we use it no overflow
-     * happens. 'uint64_t' cast is there just to prevent undefined behavior on
-     * overflow */
-    int64_t maxincr = (uint64_t)max - value;
-    int64_t minincr = min - value;
+    /* max/min and value are signed integers but to avoid undefined behavior
+     * we temporarily cast them to unsigned integers before subtracting. */
+    int64_t maxincr = (int64_t)((uint64_t)max - (uint64_t)value);
+    int64_t minincr = (int64_t)((uint64_t)min - (uint64_t)value);
```

Comment:
The cast to signed can be implicit, for brevity?

```suggestion
    int64_t maxincr = (uint64_t)max - (uint64_t)value;
    int64_t minincr = (uint64_t)min - (uint64_t)value;
```

Reply:
I made this explicit since clang-tidy otherwise complains about it:
<img width="1068" height="151" alt="image" src="https://github.com/user-attachments/assets/b223d257-aa9f-4769-9cc7-87b828d78ed6" />

Reply:
Oh, is implicit cast different to explicit? I didn't know...

Let's keep it explicit then.

Out of curiosity, do you know where this stuff is mentioned in the C standards?

Reply:
Found it: https://en.cppreference.com/w/c/language/conversion.html#Integer_promotions

Reply:
That explicit cast specifically tells clang-tidy "i know what im doing, please stop complaining", there's no other side effect and the explicit and implicit format both get compiled down to the same byte code.